subsys/nrf_security/src/drivers/cracen/psa_driver.Kconfig

```diff
@@ -12,2688 +12,821 @@ visible if 0
 config PSA_NEED_CRACEN_CCM_AES
 	bool
 	default y
+	select PSA_ACCEL_CCM_AES_128
+	select PSA_ACCEL_CCM_AES_192
+	select PSA_ACCEL_CCM_AES_256
+	depends on PSA_WANT_AES_KEY_SIZE_128
+	depends on PSA_WANT_ALG_CCM
+	depends on PSA_WANT_KEY_TYPE_AES
+	depends on PSA_USE_CRACEN_AEAD_DRIVER
 
 config PSA_NEED_CRACEN_GCM_AES
 	bool
 	default y
+	select PSA_ACCEL_GCM_AES_128
+	select PSA_ACCEL_GCM_AES_192
+	select PSA_ACCEL_GCM_AES_256
+	depends on PSA_WANT_ALG_GCM
+	depends on PSA_WANT_KEY_TYPE_AES
+	depends on PSA_USE_CRACEN_AEAD_DRIVER
 
 config PSA_NEED_CRACEN_CHACHA20_POLY1305
 	bool
 	default y
+	select PSA_ACCEL_CHACHA20_POLY1305
+	depends on PSA_WANT_ALG_CHACHA20_POLY1305
+	depends on PSA_USE_CRACEN_AEAD_DRIVER
 
 config PSA_NEED_CRACEN_AEAD_DRIVER
 	bool
 	default y
+	depends on PSA_NEED_CRACEN_CCM_AES || PSA_NEED_CRACEN_GCM_AES || PSA_NEED_CRACEN_CHACHA20_POLY1305
 
 # CRACEN Cipher Driver
 
 config PSA_NEED_CRACEN_CTR_AES
 	bool
 	default y
+	select PSA_ACCEL_CTR_AES_128
+	select PSA_ACCEL_CTR_AES_192
+	select PSA_ACCEL_CTR_AES_256
+	depends on PSA_WANT_AES_KEY_SIZE_128 || PSA_WANT_AES_KEY_SIZE_192 || PSA_WANT_AES_KEY_SIZE_256
```

Comment:
Why does CTR_AES depend on a key size being enabled, but GCM does not?

Reply:
It is aligned with the oberon configs, I thinking behind that is to make sure it is only enabled if the supported key size is set